### DISCUSSION TITLE
Push scattered Readme's to Outline using GitLab CI

### DISCUSSION THREAD

**Discussion by tillsanders** [outline/outline#3186] [Show and tell]
Hey people!

At FinanzRitter, we have lots of repositories and lots of documentation, scattered in Readme's. That makes onboarding new developers difficult as they don't know where to find information. Also, we wanted to provide some explanations of overall concepts, how everything works together et cetera. So we're now using Outline 🎉

We have some pages that we manually edit, containing onboarding documentation, abstract concepts and ideas. And then we have some subpages which are basically mirroring Readme's from our repositories. During a release pipeline, we have an additional job which will take the Readme content and replace a given Outline page with it. The result is a nice and tightly integrate documentation for everything, while a specific documentation can continue to live where the code is. 

I wrote a small article explaining why and how we did it (Code listings inside 😉): https://dev.to/tillsanders/automatically-combine-your-scattered-documentation-with-a-few-lines-in-gitlab-ci-1mlh

It would be nice to have a similar but better integration natively in the future.
Shoutout to Outline which made this really easy to implement. Some competitors don't even have an API...

Cheers
Till

**Comment by tommoor** [comment]
Nice, and a great written article. In regards to collaborative editing – it's currently a technical limitation as edits via the API would have to simulate a user editing the document in a client in order for them to be merged correctly. It's possible but as there were very few people using the `documents.update` endpoint we've avoided the work up until now.

**Reply by tillsanders** [reply]
I see, thanks for the explanation! For our use-case it would be enough to still be able to overwrite the text completely. What are your plans regarding the collaborative editing feature? Will it eventually become the standard? If so, will the API still support our use-case in the future?

**Reply by tommoor** [reply]
Yes and yes - this is one of the blockers for enabling for everyone
